### DISCUSSION TITLE
Gruntwork Pipeline alerts when deployments fail, and rollbacks?

### DISCUSSION THREAD

**Discussion by iantybo** [gruntwork-io/knowledge-base#452] [Help] [answered]
If I build Gruntwork Pipelines in my own environment, but I cannot use Slack, how can I set up other alerting mechanisms like an email via SNS or a Teams message? 

Also, how do I perform a rollback when things fail?

---

<ins datetime="2022-06-09T14:08:49Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108734">Tracked in ticket #108734</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
> how can I set up other alerting mechanisms like an email via SNS or a Teams message?

Gruntwork Pipelines relies on the associated CI servers to handle alerting. While our standard templates only support Slack, you can install additional/other integrations if the CI server supports it. For example, if you are using CircleCI, you can add https://circleci.com/developer/orbs/orb/opub/ms-teams-orb into the `.circleci/config.yaml` to implement alerting to MS Teams.

> Also, how do I perform a rollback when things fail?

Gruntwork Pipelines currently does not offer native support for a rollback. The primary reason for this is because Terraform doesn't really support a native rollback mechanism. The best way to roll back is to roll forward the deployment by pushing a revert commit using [git revert](https://git-scm.com/docs/git-revert).
